store/sql_channel_store.go

```diff
@@ -679,3 +679,25 @@ func (s SqlChannelStore) UpdateNotifyLevel(channelId, userId, notifyLevel string
 
 	return storeChannel
 }
+
+func (s SqlChannelStore) GetAllChannels(teamId string) StoreChannel {
+	storeChannel := make(StoreChannel)
+
+	go func() {
+		result := StoreResult{}
+
+		var data []*model.Channel
+		_, err := s.GetReplica().Select(&data, "SELECT * FROM Channels WHERE TeamId = :TeamId AND DeleteAt = 0", map[string]interface{}{"TeamId": teamId})
```

Comment:
If your worried about data persistence, your not going to be using this feature. So I don't think so.